usr/lib/common/asn1.c

```diff
@@ -2892,13 +2929,20 @@ CK_RV der_decode_ECPublicKey(CK_BYTE *data,
         TRACE_DEVEL("build_attribute failed\n");
         goto cleanup;
     }
-    // build ec-point attribute
-    rc = build_attribute(CKA_EC_POINT, point, point_len, &point_attr);
+    /* build ec-point attribute as BER encoded OCTET STRING */
+    rc = ber_encode_OCTET_STRING(FALSE, &ecpoint, &ecpoint_len,
+                                 point, point_len);
+    if (rc != CKR_OK) {
+        TRACE_DEVEL("ber_encode_OCTET_STRING failed\n");
+        goto cleanup;
+    }
+    rc = build_attribute(CKA_EC_POINT, ecpoint, ecpoint_len, &point_attr);
     if (rc != CKR_OK) {
         TRACE_DEVEL("build_attribute failed\n");
         goto cleanup;
     }
 
+    free(ecpoint);
```

Comment:
Check "if (ecpoint)" like in line 2754 and 2768 before free()?

Reply:
Well, ecpoint is always non NULL here. ber_encode_OCTET_STRING sets it. Either ber_encode_OCTET_STRING retutrns an error, then we don't reach here, otherwise it is set, and we can free it unconditionally.

BTW:  the NULL check before free() is superfluous anyway. free() can be called with a NULL pointer, it is just a NOP then, see man page of free.